From 2f8dec8bab9efbd0904829a4b9f7cfc79832b686 Mon Sep 17 00:00:00 2001
From: Yurii Rashkovskii <me@yrashk.com>
Date: Thu, 1 Mar 2018 11:33:05 +0700
Subject: [PATCH] Problem: sit-doc doesn't rewrite URLs of images in <img>

This results in broken images and reload flickering comparing
to images passed in markdown.

Solution: handle <img> tags
---
 sit-web/webapp/src/sit-doc/sit-doc.html | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/sit-web/webapp/src/sit-doc/sit-doc.html b/sit-web/webapp/src/sit-doc/sit-doc.html
index 3d71565..5461fc3 100644
--- a/sit-web/webapp/src/sit-doc/sit-doc.html
+++ b/sit-web/webapp/src/sit-doc/sit-doc.html
@@ -108,8 +108,10 @@
       }
 
       processMarkdownAndImages(md) {
-        return this.processMarkdown(md.replace(/\!(\[.*\])\(([^\/.].*)\)/g,
-          "!$1(/documentation/$2)"));
+        return this.processMarkdown(
+            md.replace(/\!(\[.*\])\(([^\/.].*)\)/g, "!$1(/documentation/$2)")
+                        .replace(/<img src="(?!http:\/\/|https:\/\/)([^\/].*)"/g,
+                        '<img src="https://raw.githubusercontent.com/sit-it/sit/master/doc/$1"'));
       }
 
 
-- 
2.15.0

